File: rust/src/models/create_o_auth_app_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CreateOAuthAppRequest {
    pub name: String,

    pub redirect_uris: Vec<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub scopes: Option<Vec<String>>,
}

impl CreateOAuthAppRequest {
    pub fn new(name: String, redirect_uris: Vec<String>) -> Self {
        Self {
            name,
            redirect_uris,
            scopes: None,
        }
    }
}

File: typescript/src/models/createOAuthAppRequest.ts
// this file is @generated

export interface CreateOAuthAppRequest {
  name: string;

  redirectUris: string[];

  scopes?: string[] | null;
}

export const CreateOAuthAppRequestSerializer = {
  _fromJsonObject(object: any): CreateOAuthAppRequest {
    return {
      name: object["name"],
      redirectUris: object["redirect_uris"],
      scopes: object["scopes"],
    };
  },

  _toJsonObject(self: CreateOAuthAppRequest): any {
    return {
      name: self.name,
      redirect_uris: self.redirectUris,
      scopes: self.scopes,
    };
  },
};
